### PR TITLE
Refactoring algebra tests

### PR DESCRIPTION
**Pull request by PNieck**
## Description

This PR contains refactoring of algebra tests.

## Motivation and Context

Making these tests much more clear and easier to read.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (list targets here). `armv7a9-zynq7000-qemu`

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [x] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: algebra/tests/buffs.h [hunk 1]
```diff
@@ -284,4 +291,143 @@ static const float buffs_invK[] = {
...
+static const quat_t QI = { .a = 0.0f, .i = 1.0f, .j = 0.0f, .k = 0.0f };
+static const quat_t QJ = { .a = 0.0f, .i = 0.0f, .j = 1.0f, .k = 0.0f };
+static const quat_t QK = { .a = 0.0f, .i = 0.0f, .j = 0.0f, .k = 1.0f };
+
+static const quat_t Q0 = { .a = 0.0f, .i = 0.0f, .j = 0.0f, .k = 0.0f };
+
+/* Quaternion without zero and one */
+static const quat_t Q1 = { .a = 2.0f, .i = 2.0f, .j = 2.0f, .k = 2.0f };
+
+/* Q3 must be not normalized */
```
Comment (niewim19) [inline algebra/tests/buffs.h:342]:
``` suggestion
/* Q3 norm must differ from 1 */
```
Are there any remarks about `Q2` it looks like it was omitted in comment

File: algebra/tests/buffs.h [hunk 2]
```diff
@@ -284,4 +291,143 @@ static const float buffs_invK[] = {
...
+static const quat_t Q2 = { .a = 3.0f, .i = 0.0f, .j = 6.0f, .k = 1.0f };
+static const quat_t Q3 = { .a = 1.0f, .i = 2.0f, .j = 1.0f, .k = 5.0f };
+
+/* Correct euler angels corresponding to quaternion Q3 (normalized) */
```
Comment (niewim19) [inline algebra/tests/buffs.h:346]: Please mention the correct order of rotations (from what I remember we use _ZYX_ in out implementation) as there are multiple triplets of angles that match that quaternion `Q3` depending on rotation order convention

File: algebra/tests/buffs.h [hunk 3]
```diff
@@ -284,4 +291,143 @@ static const float buffs_invK[] = {
...
+static const float Q3roll = 0.5880026;
+static const float Q3pitch = -0.6195209;
+static const float Q3yaw = 2.5535901;
+
+static const quat_t Q2timesQ3 = { .a = -8.0f, .i = 35.0f, .j = 11.0f, .k = 4.0f };
+static const quat_t Q2sandQ3 = { .a = 46.0f, .i = 118.0f, .j = 116.0f, .k = -190.0f };
+
+static const quat_t Q4 = { .a = 815.23f, .i = -818.07f, .j = -451.47f, .k = -546.79f };
+static const quat_t Q5 = { .a = 334.23f, .i = -822.81f, .j = 349.42f, .k = 548.18f };
```
Comment (niewim19) [inline algebra/tests/buffs.h:355]: Please add description for these quaternions, especially those that represent products and sandwich

File: algebra/tests/buffs.h [hunk 4]
```diff
@@ -284,4 +291,143 @@ static const float buffs_invK[] = {
...
+
+/* Correct euler angels corresponding to quaternion Q5 (normalized) */
```
Comment (niewim19) [inline algebra/tests/buffs.h:357]: Please add correct rotations order for that angles triplet

File: algebra/tests/buffs.h [hunk 5]
```diff
@@ -284,4 +291,143 @@ static const float buffs_invK[] = {
...
+static const float Q5roll = -2.7342767;
+static const float Q5pitch = 1.2155062;
+static const float Q5yaw = -0.5178249;
+
+static const quat_t Q4timesQ5 = { .a = 56850.1358f, .i = -1000630.3952f, .j = 1032316.741f, .k = -393184.8904f };
+static const quat_t Q4sandQ5 = {
```
Comment (niewim19) [inline algebra/tests/buffs.h:363]: Missing descriptions

File: algebra/tests/buffs.h [hunk 6]
```diff
@@ -284,4 +291,143 @@ static const float buffs_invK[] = {
...
+static const vec_t V9 = { .x = -261.48f, .y = 731.11f, .z = -919.51f };
+
+/* Normalized rotation quaternions */
+static const quat_t Q8 = { .a = 0.1825742f, .i = 0.3651484f, .j = 0.5477226f, .k = 0.7302967f };
+
+/* Vectors after rotation using Q8 */
+static const vec_t V8rotQ8 = { .x = 1.8f, .y = 2.0f, .z = 2.6f };
+static const vec_t V9rotQ8 = { .x = -402.5060f, .y = -1031.03f, .z = 472.6079f };
+
+/* Data used in `quat_rotQuat` function */
```
Comment (niewim19) [inline algebra/tests/buffs.h:386]:
```suggestion
/* Data used in `quat_rotQuat` function tests */
```

File: algebra/tests/buffs.h [hunk 7]
```diff
@@ -284,4 +291,143 @@ static const float buffs_invK[] = {
...
+static const vec_t V10 = { .x = 5.0f, .y = 2.0f, .z = 3.0f };
+
+/* Angle in radians */
+static const float Angle = 1;
+
+/* Quaternion of rotation about `Angle` and along `V3` axis */
+static const quat_t Q9 = { .a = 0.8775826f, .i = 0.3888655f, .j = 0.155546f, .k = 0.2333193f };
+
+/* Vector nearly antiparallel to V10 */
```
Comment (niewim19) [inline algebra/tests/buffs.h:395]:
``` suggestion
/* Vector nearly antiparallel to V10. Antiparallel = parallel, but with opposite directions */
```

File: algebra/tests/buffs.h [hunk 8]
```diff
@@ -284,4 +291,143 @@ static const float buffs_invK[] = {
...
+static const vec_t V11 = { .x = -4.52f, .y = -1.44f, .z = -3.0f };
+
+/* Vector nearly parallel to V10 */
+static const vec_t V12 = { .x = 7.68f, .y = 2.26f, .z = 5.0f };
+
+/* Vectors for `quat_frameRot` tests */
```
Comment (niewim19) [inline algebra/tests/buffs.h:401]:
``` suggestion
/* Data sets for `quat_frameRot` tests */
```

**Review by niewim19 — COMMENTED** [review]
### Moving `buffs` up the direcotry tree

Maybe it is possible to separate this header into three smaller and stored in proper test directories (three files in total). I know that there are some common values (such zero vector) but maybe it is not as many of them so they can be stored twice?

### Naming

Please change filenames in `matrix` tests module:
 - `addition_and_subtraction.c` to something shorter like `add_sub.c`)
 - `basic_interactions.c` - maybe just `basics.c`
 - `matrix_multiplication` - we are in `matrix` tests module so maybe just `multiplication.c`?
 - `tools_tests.c` -> `tools.c`. I do not know how this will work with the `tools` higher in the directories

in `quat` tests module
 - `quat_various.c` -> `basics.c`
 
 - `vectors_tests.c` -> `basics.c` or just `tests.c`

File: algebra/tests/buffs.h (hunk 1 above)
Comment (PNieck) [reply algebra/tests/buffs.h:342]: There is no remarks on Q2, because there is nothing special in this matrix

Comment (niewim19) [reply algebra/tests/buffs.h:342]: So if i would like to change it with all zeroes wouldn`t that make a difference? :) If there are really no remarks about this matrix, then _IT IS_ the remark - that it is _matrix with some arbitrary values_

**Review by niewim19 — APPROVED** [review]
LGTM

**Pull request merged**